src/sql/workbench/services/metadata/common/interfaces.ts

```diff
@@ -0,0 +1,43 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import { createDecorator } from 'vs/platform/instantiation/common/instantiation';
+
+
+export const SERVICE_ID = 'serverMetadataService';
+export const IServerMetadataService = createDecorator<IServerMetadataService>(SERVICE_ID);
+
+export interface IServerMetadataService {
+	_serviceBrand: undefined;
```

Comment:
what's this used for? I did a quick search in the PR and couldn't find where it's used

Reply:
I'm not entirely sure what `_serviceBrand` is for. It looks like it was something that was brought in through a VS Code merge 4 years ago according to git blame.

There's a few other places that use it, but it also goes unused. These are just 2 examples, but there seem to be a lot of services that have it and don't use it:

https://github.com/microsoft/azuredatastudio/blob/07f32dd0fb652eb418d51b6f773097afb62a854e/src/sql/workbench/services/tableDesigner/browser/tableDesignerService.ts#L30

https://github.com/microsoft/azuredatastudio/blob/07f32dd0fb652eb418d51b6f773097afb62a854e/src/sql/workbench/services/dashboard/browser/newDashboardTabDialogService.ts#L15

Reply:
It's used to mark types that are services when doing dependency injection through the instantiation service. So it never has/doesn't need a value since the existence of the property itself is enough.

https://github.com/microsoft/azuredatastudio/blob/76dd6989aed8e4d8bf79c7eeb61049b9f1c95f76/src/vs/platform/instantiation/common/instantiation.ts#L25